### ISSUE TITLE
[jit] self getting overwritten when model is saved

### ISSUE THREAD

**Issue by suo** [pytorch/pytorch#17147]
From: https://discuss.pytorch.org/t/trouble-with-reshape-and-jit-trace/36346/3

```
#!/usr/bin/ipython3

import torch

class Example(torch.nn.Module):
def init(self):
super(Example, self).init()

def forward(self, x):
    s = x.shape
    b = x.view(x.shape[0],x.shape[1]//2,2)
    accum = b.sum(1, keepdim=True)
    b = b * accum
    return b.view(*s)
class ExampleNested(torch.nn.Module):
def init(self):
super(ExampleNested, self).init()
self.ex = Example()
self.lin = torch.nn.Linear(4,4)

def forward(self, x):
    x = self.ex(x)
    x = self.lin(x)
    return x
a = torch.randn(4,4)

example = ExampleNested()
traced = torch.jit.trace(example, a)
traced.save(“trace.tmp”)
```

gives
```
op_version_set = 0
def forward(self,
x: Tensor) -> Tensor:
_0 = ops.prim.NumToTensor(torch.size(x, 0))
_1 = int(_0)
_2 = ops.prim.NumToTensor(torch.size(x, 1))
_3 = int(_2)
_4 = ops.prim.NumToTensor(torch.size(x, 0))
_5 = int(_4)
self = ops.prim.NumToTensor(torch.size(x, 1))
_6 = [_5, int(torch.div(self, CONSTANTS.c0)), 2]
b_1 = torch.view(x, _6)
accum = torch.sum(b_1, [1], True)
b = torch.mul(b_1, accum)
input = torch.view(b, [_1, _3])
_7 = torch.addmm(self.lin.bias, input, torch.t(self.lin.weight), beta=1, alpha=1)
return _7
```

note the line `self = ops.prim.NumToTensor(torch.size(x, 1))`.

**Comment by zdevito** [comment]
I can't reproduce this off of recent master.  I get the following:

```
op_version_set = 0
def forward(self,
    x: Tensor) -> Tensor:
  _0 = ops.prim.NumToTensor(torch.size(x, 0))
  _1 = int(_0)
  _2 = ops.prim.NumToTensor(torch.size(x, 1))
  _3 = int(_2)
  _4 = ops.prim.NumToTensor(torch.size(x, 0))
  _5 = ops.prim.NumToTensor(torch.size(x, 1))
  _6 = [int(_4), int(torch.div(_5, CONSTANTS.c0)), 2]
  b = torch.view(x, _6)
  accum = torch.sum(b, [1], True)
  b0 = torch.mul(b, accum)
  input = torch.view(b0, [_1, _3])
  _7 = torch.addmm(self.lin.bias, input, torch.t(self.lin.weight), beta=1, alpha=1)
  return _7
```

There was a bug fix that went in between 1.0.0 and 1.0.1 that fixed a similar bug. It is possible this only happens on the older version.

**Comment by driazati** [comment]
Bug fix is in #15318

**Comment by suo** [comment]
Going to close this as it appears it to be fixed. Feel free to reopen it if you are still seeing issues on 1.0.1
